unity_breadcrumbs/src/DecisionBreadcrumb.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\unity_breadcrumbs;
+
+/**
+ * @file
+ * Generates the breadcrumb trail for content including:
+ * - Decision
+ *
+ * In the format:
+ * > Home
+ * > Decisions
+ * > current-page-title
+ *
+ * > <front>
+ * > /decisions
+ * > /current-page-title
+ */
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Controller\TitleResolverInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * {@inheritdoc}
+ */
+class DecisionBreadcrumb implements BreadcrumbBuilderInterface {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Node object, or null if on a non-node page.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  protected $node;
+
+  /**
+   * The title resolver.
+   *
+   * @var \Drupal\Core\Controller\TitleResolverInterface
+   */
+  protected $titleResolver;
+
+  /**
+   * RequestStack service object.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $request;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, TitleResolverInterface $title_resolver, RequestStack $request) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->titleResolver = $title_resolver;
+    $this->request = $request;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('title_resolver'),
+      $container->get('request_stack')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+    $route_name = $route_match->getRouteName();
+    if ($route_name == 'entity.node.canonical') {
+      $this->node = $route_match->getParameter('node');
+    }
+
+    if ($route_name == 'entity.node.preview') {
+      $this->node = $route_match->getParameter('node_preview');
+    }
+
+    if (!empty($this->node)) {
```

Comment:
I haven't tried it but this empty() check might be redundant as there's a check below to see if $this->node is an instance of NodeInterface. Might be worth a look to see if that extra condition can be removed and make it slightly clearer to read but if not, it's perfectly fine to go as it is.

Reply:
Yes, you are right. Taking out the empty check gives the same result. I have fixed this throughout the module.